nemo/core/classes/modelPT.py

```diff
@@ -981,6 +985,44 @@ def cfg(self, cfg):
         self._cfg = cfg
         self._set_hparams(cfg)
 
+    @staticmethod
+    def update_model_dataclass(model_cls: ModelPTConfig, update_cfg: DictConfig):
+        """
+        Accepts a ModelPT dataclass and a DictConfig which mirrors the structure of the
+        dataclass, so as to update the dataclass with the values from the DictConfig.
+
+        Essentially, the update_cfg is obtained via hydra overrides or via manual construction,
+        and used to override the default values set in the dataclass.
+
+        # Usage
+        @hydra_runner(config_path="conf", config_name="config")
+        def main(cfg):
+            # Create the data class with valid defaults
+            model_dataclass = ModelPTConfig()
+
+            # Override the defaults with hydra / external DictConfig which mirrors the structure
+            cfg = ModelPT.update_model_dataclass(model_dataclass, cfg)
```

Comment:
Why do we need these extra steps? I was thinking that our example scripts would stay exactly the same, and the only thing that would be changing for users is there would be less configurations in the config files as everything needed (except paths to data, etc) would be configured from the dataclass.

Reply:
What is the difference between `update_model_dataclass` and `update_model_config`? Aren't those two covering the same functionality?

Reply:
Updated documentation, but basically `update_model_dataclass ` is a forwarding of the `update_model_config` for easy API. Can be changed if it seems we are polluting the ModelPT namespace.

Reply:
I don't understand why this method is necessary. It literally just calls ```update_model_config```.
Some people will do things like:
```
mymodel.update_model_dataclass(whatever)
```
and expect ``mymodel`` to change (and, of course, it won't). I would remove this from ``ModelPT``